framework/moose.mk

```diff
@@ -38,7 +38,7 @@ pcre_deps      := $(patsubst %.cc, %.$(obj-suffix).d, $(pcre_srcfiles)) \
 #
 # hit (new getpot parser)
 #
-hit_DIR       := $(FRAMEWORK_DIR)/contrib/hit
+hit_DIR       ?= $(MOOSE_DIR)/moosetools/contrib/hit
```

Comment:
Yes, I agree that this needs to be renamed. The build problems are now gone. It appears that the unit tests are now broken though.

Reply:
Yes, I still need to check for the submodule not being inited.

Reply:
That is probably just too much being removed. All of the unit tests have been moved to moosetools, except for the use of fparser. I didn't include the fparser in moosetools, so those tests need to stay in place.

Reply:
The issue here is that you've deleted or moved some of the types. "gtest" probably shouldn't move because we need a way to unit test MOOSE. However, I do see the dilemma of only having "gtest" in MOOSE. That would mean we can't unit test the HIT parser...

Reply:
I'm not sure what the goal is here. How would you expect MOOSE to work without HIT (in the moosetools repo)? I don't see how we can make this a soft dependency. If it's not a soft dependency, then we probably don't need to cleanup the unit tests (assume that gtest and HIT are always available).